app/assets/javascripts/preload-application-data.js

```diff
@@ -1,12 +1,42 @@
 (function() {
+  var ps = require("preload-store").default;
   var preloadedDataElement = document.getElementById("data-preloaded");
 
   if (preloadedDataElement) {
-    var ps = require("preload-store").default;
     var preloaded = JSON.parse(preloadedDataElement.dataset.preloaded);
 
     Object.keys(preloaded).forEach(function(key) {
       ps.store(key, JSON.parse(preloaded[key]));
     });
   }
+
+  var setupData = document.getElementById("data-discourse-setup").dataset;
+
+  Discourse.CDN = setupData.cdn;
+  Discourse.BaseUrl = setupData.baseUrl;
```

Comment:
Hmm is there something that automatically coverts the key from snake case to camel case?

Reply:
Yes, it is part of the `dataset` property magic: https://developer.mozilla.org/en-US/docs/Web/API/HTMLElement/dataset

> The name of a custom data attribute in JavaScript is the name of the same HTML attribute but in camelCase and with no dashes, dots, etc.

Reply:
Nice :+1: learnt something new today